pkg/types/automation/interfaces.go

```diff
@@ -0,0 +1,79 @@
+package automation
+
+import (
+	"context"
+	"io"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+)
+
+// UpkeepStateStore is the interface for managing upkeeps final state in a local store.
+type UpkeepStateStore interface {
+	UpkeepStateUpdater
+	UpkeepStateReader
+	Start(context.Context) error
```

Comment:
We can avoid `Start()` and `Close()` functions for all interfaces, as they are not really part of the interfaces (at least not the logical interface) and are only called from within the repo with the implementations